src/scalar.rs

```diff
@@ -568,7 +571,48 @@ impl Zeroize for Scalar {
     }
 }
 
+#[cfg(all(feature = "ff", feature = "rand_core"))]
+impl Field for Scalar {
+    const ZERO: Self = Self::ZERO;
+    const ONE: Self = Self::ONE;
+
+    fn random(mut rng: impl RngCore) -> Self {
+        // NOTE: this is duplicated due to different `rng` bounds
+        let mut scalar_bytes = [0u8; 64];
+        rng.fill_bytes(&mut scalar_bytes);
+        Self::from_bytes_mod_order_wide(&scalar_bytes)
+    }
+
+    fn square(&self) -> Self {
+        self * self
+    }
+
+    fn double(&self) -> Self {
+        self + self
+    }
+
+    fn invert(&self) -> CtOption<Self> {
+        CtOption::new(self.invert(), Choice::from(1u8))
+    }
+
+    fn sqrt_ratio(_num: &Self, _div: &Self) -> (Choice, Self) {
+        // TODO: can't use `ff::helpers::sqrt_ratio_generic` because `Scalar` isn't a `PrimeField`
```

Comment:
The documentation of `Scalar` indicates that its arithmetic is modulo `ell`, the prime order of both Ristretto and the Ed25519 basepoint. So we can `impl PrimeField for Scalar`.